wpilibc/src/test/native/cpp/AnalogPotentiometerTest.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+#include <hal/HAL.h>
+
+#include "frc/AnalogPotentiometer.h"
+#include "frc/simulation/AnalogInputSim.h"
+#include "frc/simulation/RoboRioSim.h"
+#include "gtest/gtest.h"
+
+namespace frc {
+using namespace frc::sim;
+TEST(AnalogPotentiometerTest, InitializeWithAnalogInput) {
+  HAL_Initialize(500, 0);
+
+  AnalogInput ai(0);
+  AnalogPotentiometer pot(&ai);
+  AnalogInputSim sim(ai);
+
+  RoboRioSim::ResetData();
+  sim.SetVoltage(4.0);
+  EXPECT_EQ(0.8, pot.Get());
+}
+
+TEST(AnalogPotentiometerTest, InitializeWithAnalogInputAndScale) {
+  HAL_Initialize(500, 0);
+
+  AnalogInput ai(0);
+  AnalogPotentiometer pot(&ai, 270.0);
+  RoboRioSim::ResetData();
+  AnalogInputSim sim(ai);
+
+  sim.SetVoltage(5.0);
+  EXPECT_EQ(270.0, pot.Get());
+
+  sim.SetVoltage(2.5);
+  EXPECT_EQ(135, pot.Get());
+
+  sim.SetVoltage(0.0);
+  EXPECT_EQ(0.0, pot.Get());
+}
+
+TEST(AnalogPotentiometerTest, InitializeWithChannel) {
+  HAL_Initialize(500, 0);
```

Comment:
Is this `HAL_Initialize()` call necessary? Some tests don't have one.

Reply:
Realistically they should have it to ensure they don't segfault if the order of the tests change. I would vote for a global test fixture that initializes the hal and resets all sim data so that every test can be "clean", but that seems like a different PR.

Reply:
In C++, the reason not all tests have this is main() calls HAL_Initialize. So its not actually necessary anywhere in C++ tests.

Reply:
Thats a good point. I'll remove them all from my tests, and can make a follow up PR for removing any remaining. Meshes well with #3495.

I still would vote for the resetting sim data with a [global](https://github.com/google/googletest/blob/master/docs/advanced.md#global-set-up-and-tear-down) setup / tear down function